.github/workflows/release_batch_transformer.yml

```diff
@@ -0,0 +1,218 @@
+name: Release batch transformer 
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  release-batch:
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Github
+        uses: actions/checkout@v2
+
+      - uses: coursier/cache-action@v6
+
+      - name: Set up JDK 1.11 for batch transformer
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.11
+          name: build
+
+      - name: Get current SBT version
+        id: ver
+        run: |
+          export PROJECT_VERSION=$(sbt version -Dsbt.log.noformat=true | perl -ne 'print "$1\n" if /info.*(\d+\.\d+\.\d+[^\r\n]*)/' | tail -n 1 | tr -d '\n')
+          echo "::set-output name=project_version::$PROJECT_VERSION"
+
+      - name: Compare SBT version with git tag
+        run: |
+          tag=${GITHUB_REF##*/}
+          project_version=${{ steps.ver.outputs.project_version }}
+          if [[ "${tag}" = "${project_version}" ]]; then
+            echo "Tag version (${tag}) matches project version (${project_version}). Deploying!"
+          else
+            echo "Tag version (${tag}) doesn't match version in scala project (${project_version}). Aborting!"
+            exit 1
+          fi
+
+      - name: Build artifact
+        run: sbt "project transformerBatch" "set test in assembly := {}" assembly
+
+      - name: Set environmental variables
+        id: vars
+        run: |
+          export SOURCE_DIR=modules/transformer-batch/target/scala-2.12/assembled-jar
+          export DEST_DIR=4-storage/transformer-batch
+          echo "::set-output name=source_dir::$SOURCE_DIR"
+          echo "::set-output name=dest_dir::$DEST_DIR"
+
+      - name: Deploy to eu-west-1
+        uses: jakejarvis/s3-sync-action@master
+        with:
+          args: --acl public-read --follow-symlinks
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_KEY }}
+          AWS_S3_BUCKET: snowplow-hosted-assets
+          AWS_REGION: eu-west-1
```

Comment:
It seems like only region is changed in the repeated sections here. We might use [the matrix strategy](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) to define region as variable. We've used similar technique in [here](https://github.com/snowplow/snowplow-rdb-loader/blob/develop/.github/workflows/ci.yml#L66-L73).

Reply:
I thought about that, but if we use matrix, would that mean whole job (with assembly) needs to be repeated for every region? Unless there is a way to use a matrix on a step level, instead of a job level. 

Assuming matrix can only be used on a job level, what options do we have:

* repeat whole job for each region. I can test it to see how much time it takes. Maybe it's quick and we don't have to worry about time. Especially when we run it in parallel.
* split job to sth like `assemble` + `publish` and use matrix only in `publish`. Then we would have to use output of `assemble` in `publish`, not sure it's possible...? 

  WDYT?

Reply:
Ah, you are right. I guess there is no neat solution for this problem. I think it is okay to leave it like that.